src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
```

Comment:
PullProofAI

The `HttpExceptionFilter` class's `catch` method seems to have a complex error handling logic that can be simplified and improved for better readability and maintainability. Consider refactoring this part of the code.

```diff
-     let err = exception['response'];
-     err = err.response ? err.response : JSON.stringify(err);
+     const err = exception.getResponse();
```

```diff
-     const error =
-       typeof exception.getResponse() === 'string'
-         ? exception.getResponse()
-         : exception.getResponse() &&
-             typeof exception.getResponse()['message'] === 'object'
-           ? exception.getResponse()['message'].join(', ')
-           : exception.getResponse()['message'];
+     const error = typeof err === 'string' ? err : err && typeof err.message === 'object' ? err.message.join(', ') : err.message;
```

This refactoring simplifies the error handling logic by directly using the `exception.getResponse()` result and improves code readability.

### End of Review


<!-- This is an auto-generated comment by PullProofAI -->